minigu-test/src/slt_adapter.rs

```diff
@@ -0,0 +1,354 @@
+use std::sync::{Arc, Mutex};
+
+use minigu::common::data_type::LogicalType;
+use minigu::database::{Database, DatabaseConfig};
+use minigu::error::Error as MiniGuError;
+use minigu::session::Session;
+use sqllogictest::{ColumnType, DB, DBOutput};
+
+/// Enhanced ColumnType for MiniGU that supports vertex and edge types
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum MiniGuColumnType {
+    /// Text, varchar results
+    Text,
+    /// Integer results
+    Integer,
+    /// Floating point numbers
+    FloatingPoint,
+    /// Boolean results
+    Boolean,
+    /// Vertex type (graph-specific)
+    Vertex,
+    /// Edge type (graph-specific)
+    Edge,
+    /// Any other type
+    Any,
+}
+
+impl ColumnType for MiniGuColumnType {
+    fn from_char(value: char) -> Option<Self> {
+        match value {
+            'T' => Some(Self::Text),
+            'I' => Some(Self::Integer),
+            'R' => Some(Self::FloatingPoint),
+            'V' => Some(Self::Vertex),
+            'E' => Some(Self::Edge),
+            'B' => Some(Self::Boolean),
+            _ => Some(Self::Any),
+        }
+    }
+
+    fn to_char(&self) -> char {
+        match self {
+            Self::Text => 'T',
+            Self::Integer => 'I',
+            Self::FloatingPoint => 'R',
+            Self::Boolean => 'B',
+            Self::Vertex => 'V',
+            Self::Edge => 'E',
+            Self::Any => '?',
+        }
+    }
+}
+
+/// Convert LogicalType to MiniGuColumnType
+impl From<&LogicalType> for MiniGuColumnType {
+    fn from(logical_type: &LogicalType) -> Self {
+        match logical_type {
+            LogicalType::String => Self::Text,
+            LogicalType::Int8
+            | LogicalType::Int16
+            | LogicalType::Int32
+            | LogicalType::Int64
+            | LogicalType::UInt8
+            | LogicalType::UInt16
+            | LogicalType::UInt32
+            | LogicalType::UInt64 => Self::Integer,
+            LogicalType::Float32 | LogicalType::Float64 => Self::FloatingPoint,
+            LogicalType::Boolean => Self::Boolean,
+            LogicalType::Vertex(_) => Self::Vertex,
+            LogicalType::Edge(_) => Self::Edge,
+            LogicalType::Record(_) => Self::Any,
+            LogicalType::Null => Self::Any,
+        }
+    }
+}
+
+/// Session wrapper for MiniGU that maintains session state across multiple SQL statements
+#[derive(Clone)]
+pub struct SessionWrapper {
+    database: Arc<Database>,
+    session: Option<Arc<Mutex<Session>>>,
+}
+
+impl SessionWrapper {
+    /// Create new SessionWrapper with a fresh database
+    pub fn new() -> Result<Self, MiniGuError> {
```

Comment:
It is better to create a `SessionWrapper` from a given `Session`. And `Database` should be created by the caller. This enables the caller to configure the database before testing.